src/loan/terms/simple/loan/pwn_simple_loan.cairo

```diff
@@ -81,100 +146,704 @@ mod PwnSimpleLoan {
             caller_spec: CallerSpec,
             extra: Option<Array<felt252>>
         ) -> felt252 {
-            0
+            let caller = starknet::get_caller_address();
+            if (!self
+                .hub
+                .read()
+                .has_tag(proposal_spec.proposal_contract, pwn_hub_tags::LOAN_PROPOSAL)) {
+                // @note: we shall move these error in a common place?
```

Comment:
what do you mean for 'common place?'

Reply:
I thought I was a bid odd to get the error from a different module, and we can have errors at one common place